eclib/JModel.ec

```diff
@@ -515,91 +924,71 @@ abbrev [-printing] VAESENCLAST      = AESENCLAST.
 
-abbrev [-printing] (\vadd8u256)  (w1 w2:W256.t) = VPADD_32u8 w1 w2.
-abbrev [-printing] (\vadd16u256) (w1 w2:W256.t) = VPADD_16u16 w1 w2.
-abbrev [-printing] (\vadd32u256) (w1 w2:W256.t) = VPADD_8u32 w1 w2.
-abbrev [-printing] (\vadd64u256) (w1 w2:W256.t) = VPADD_4u64 w1 w2.
+abbrev [-printing] const_rotate8_256 =
+  W256.of_int 6355432118420048154175784972596847518577147054203239762089463134348153782275.
 
-abbrev [-printing] (\vsub8u256)  (w1 w2:W256.t) = VPSUB_32u8 w1 w2.
-abbrev [-printing] (\vsub16u256) (w1 w2:W256.t) = VPSUB_16u16 w1 w2.
-abbrev [-printing] (\vsub32u256) (w1 w2:W256.t) = VPSUB_8u32 w1 w2.
-abbrev [-printing] (\vsub64u256) (w1 w2:W256.t) = VPSUB_4u64 w1 w2.
+abbrev [-printing] const_rotate16_256 =
+  W256.of_int 5901373100945378232718128989223044758631764214521116316503579100742837863170.
 
-abbrev [-printing] (\vmul8u256)  (w1 w2:W256.t) = VPMUL_32u8 w1 w2.
-abbrev [-printing] (\vmul16u256) (w1 w2:W256.t) = VPMUL_16u16 w1 w2.
-abbrev [-printing] (\vmul32u256) (w1 w2:W256.t) = VPMUL_8u32 w1 w2.
-abbrev [-printing] (\vmul64u256) (w1 w2:W256.t) = VPMUL_4u64 w1 w2.
+abbrev [-printing] const_rotate24_256 =
+  W256.of_int 5454353864746073763129182254217446065883741921538078285974850505695092212225.
 
+lemma iteri_red n (opr : int -> 'a -> 'a) x : 
+  0 < n => iteri n opr x = opr (n-1) (iteri (n-1) opr x).
+proof. smt (iteriS). qed.
 
-(* ------------------------------------------------------------------- *)
+lemma rotate8_256_E w :
+  VPSHUFB_256 w const_rotate8_256 = W8u32.map (fun w => W32.rol w 8) w.
+proof. by apply: (W256.all_eq_eq _ _ _); cbv delta. qed.
 
-abbrev [-printing] set0_8   = W8.ALU.set0_8_.
-abbrev [-printing] set0_16  = W16.ALU.set0_16_.
-abbrev [-printing] set0_32  = W32.ALU.set0_32_.
-abbrev [-printing] set0_64  = W64.ALU.set0_64_.
-abbrev [-printing] set0_128 = W128.zero. 
-abbrev [-printing] set0_256 = W256.zero.
+lemma rotate16_256_E w :
+  VPSHUFB_256 w const_rotate16_256 = W8u32.map (fun w => W32.rol w 16) w.
+proof. by apply: (W256.all_eq_eq _ _ _); cbv delta. qed.
 
-(* ------------------------------------------------------------------- *)
-(* Notation used for array copy                                        *)
-op copy_8   ['a] (x:'a) = x.
-op copy_16  ['a] (x:'a) = x.
-op copy_32  ['a] (x:'a) = x.
-op copy_64  ['a] (x:'a) = x.
-op copy_128 ['a] (x:'a) = x.
-op copy_256 ['a] (x:'a) = x.
+lemma rotate24_256_E w :
+  VPSHUFB_256 w const_rotate24_256 = W8u32.map (fun w => W32.rol w 24) w.
+proof. by apply: (W256.all_eq_eq _ _ _); cbv delta. qed.
 
-(* ------------------------------------------------------------------- *)
-abbrev [-printing] adc_8  = W8.addc.
-abbrev [-printing] adc_16 = W16.addc.
-abbrev [-printing] adc_32 = W32.addc.
-abbrev [-printing] adc_64 = W64.addc.
+hint simplify (rotate8_256_E, rotate16_256_E, rotate24_256_E).
 
 (* ------------------------------------------------------------------- *)
 abbrev [-printing] sbb_8  = W8.subc.
```

Comment:
This definition (similarly to the three other ones below) must be removed as it has been reintroduced above.

Reply:
Done.